src/overlays/effects/ovl_Effect_Ss_Fhg_Flash/z_eff_ss_fhg_flash.c

```diff
@@ -1,27 +1,120 @@
 /*
  * File: z_eff_ss_fhg_flash.c
  * Overlay: ovl_Effect_Ss_Fhg_Flash
- * Description:
+ * Description: Shock and Light Ball Effect
  */
 
 #include "z_eff_ss_fhg_flash.h"
+#include "overlays/actors/ovl_Boss_Hakugin/z_boss_hakugin.h"
+#include "assets/overlays/ovl_Effect_Ss_Fhg_Flash/ovl_Effect_Ss_Fhg_Flash.h"
+
+#define rAlpha regs[0]
+#define rXZRot regs[3]
+#define rParams regs[4]
+#define rScale regs[8]
 
 #define PARAMS ((EffectSsFhgFlashInitParams*)initParamsx)
 
-s32 EffectSsFhgFlash_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsFhgFlash_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsFhgFlash_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsFhgFlash_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_Fhg_Flash_InitVars = {
     EFFECT_SS_FHG_FLASH,
     EffectSsFhgFlash_Init,
 };
 
-#endif
+u32 EffectSsFhgFlash_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsFhgFlashInitParams* initParams = PARAMS;
+    Vec3f noActorPos = { 0.0f, -1000.0f, 0.0f };
+
+    this->actor = initParams->actor;
+    Math_Vec3f_Copy(&this->velocity, &gZeroVec3f);
+    Math_Vec3f_Copy(&this->accel, &gZeroVec3f);
+    this->life = (s32)(Rand_ZeroOne() * 10.0f) + 111;
+    this->rScale = (s32)Rand_ZeroFloat(initParams->scale) + initParams->scale;
+    this->rAlpha = 255;
+    this->draw = EffectSsFhgFlash_Draw;
+    this->update = EffectSsFhgFlash_Update;
+    this->rParams = initParams->params;
+
+    if (initParams->params != FHGFLASH_SHOCK_NO_ACTOR) {
+        this->pos = noActorPos;
+        this->gfx = gFhgFlashDL;
+    } else {
+        this->pos = initParams->pos;
+        this->gfx = gFhgFlashDL;
+    }
+
+    return 1;
+}
+
+Vec3f sZeroVec = { 0.0f, 0.0f, 0.0f };
```

Comment:
```suggestion
static Vec3f sZeroVec = { 0.0f, 0.0f, 0.0f };
```
`sZeroVec` is too generic of a name and might clash with other files

Reply:
Side note, from OoT this used to just be declared on the stack in `Init`, and it still works if I make it in-function static. So I'm also going to move it into `Init`.